src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/services/EventMessageReceiverService.java

```diff
@@ -87,6 +87,20 @@ public CaseEventMessage getMessage(String messageId) {
         return mapper.mapToCaseEventMessage(messageEntity);
     }
 
+    private CaseEventMessageEntity insertMessage(CaseEventMessageEntity caseEventMessageEntity) {
+
+        final List<CaseEventMessageEntity> existingMessages =
+                repository.findByMessageId(caseEventMessageEntity.getMessageId());
+        if (existingMessages.isEmpty()) {
+            return repository.save(caseEventMessageEntity);
```

Comment:
still default? should this be using native sql as specified in the story description?

Reply:
Was so much more difficult to code with native SQL query, requiring casts of json data etc - this is much simpler and is functionally equivalent